test/transport/custom_comm/CustomComm.hpp

```diff
@@ -0,0 +1,90 @@
+// Copyright 2018 Proyectos y Sistemas de Mantenimiento SL (eProsima).
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef _TEST_TRANSPORT_SERIALCOMM_HPP_
+#define _TEST_TRANSPORT_SERIALCOMM_HPP_
+
+#include <gtest/gtest.h>
+#include <uxr/client/client.h>
+
+class CircularBuffer {
+    public:
+        CircularBuffer(size_t len) : maxlen(len), head(0), tail(0) {
+            buffer = static_cast<uint8_t*>(malloc(maxlen));
+        }
+
+        ~CircularBuffer()
+        {
+            free(buffer);
+        }
+
+        int write(uint8_t data)
+        {
+            size_t next;
+
+            next = head + 1;
+            if (next >= maxlen)
+                next = 0;
```

Comment:
```suggestion
            {
                next = 0;
            }
```
Please update every case

Reply:
Done